Sphinx correct output when disabled/down

kwikadi opened the pull request:
I had to close the other PR (#1006) because I ended up making up a new branch (because of merge conflicts and whatnot).

jiru commented on app/controllers/sentences_controller.php:
```diff
@@ -216,8 +217,21 @@ public function show($id = null)
+                    $this->Session->setFlash(__('Search was disabled. Please try again later.', true));
+                } else {
+                    $html = new HtmlHelper();
+                    $this->Session->setFlash(format(__('An error occurred while performing this function. 
```
Please use a more specific message, like `An error occurred while retrieving a random sentence.`

jiru commented on app/controllers/sentences_controller.php:
```diff
@@ -216,8 +217,21 @@ public function show($id = null)
+                    $html = new HtmlHelper();
+                    $this->Session->setFlash(format(__('An error occurred while performing this function. 
+                            If the error persists, please <a href="{}">let us know</a>.', true),
+                        $html->url(array("controller"=>"pages", "action" => "contact"))));
```
You’re messing with MVC hard here. :-)

One way to avoid this is to print the message directly from the view, instead of using flash messages. See this [example](https://github.com/Tatoeba/tatoeba2/blob/dev/app/views/transcriptions/view.ctp).

jiru commented on app/controllers/sentences_controller.php:
```diff
@@ -960,10 +974,36 @@ public function random($lang = null)
+                                    "Translation" => array(),
+
+                                    "Transcription" => array()
+                                );
```
LOL. Did you really mean to commit that?

jiru commented on app/views/helpers/sentences.php:
```diff
@@ -444,9 +444,11 @@ public function displayMainSentence(
+            $this->Menu->displayMenu(
+                $sentenceId, $user, $script, $canTranslate, $langFilter, $hasAudio
+            );
+        }
```
I don’t think you meant to commit this part.

jiru commented:
Thank you, kwikadi. Except the few things I mentioned, it’s pretty good. :-)

kwikadi commented:
I didn't quite understand the part about the transcriptions that you mentioned, but I removed the message from the controller to the view. Fixed the rest too. If there are still issues, please let me know :)

jiru commented on app/views/sentences/several_random_sentences.ctp:
```diff
@@ -102,9 +102,28 @@ $this->set('title_for_layout', $pages->formatTitle(__('Random sentences', true))
+                echo $html->tag('h2', __('Search is currently disabled, please try again later.', true));
+            } else if ($searchProblem == 'error') {
+                echo $html->tag('h2', format(__('An error occurred while fetching random sentences. 
+                        If this persists, please <a href="{}">let us know</a>.', true),
```
This string spans on two lines, which unnecessarily adds a newline and a lot of spaces between the two sentences. Please use concatenation instead:
```php
'First sentence. '. # note the leading space inside the string here
'And this is the '. # and here too
'second sentence'
```

jiru commented on app/views/sentences/show.ctp:
```diff
@@ -233,4 +234,22 @@ $navigation->displaySentenceNavigation(
+                echo $html->tag('h2', __('Search is currently disabled, please try again later.', true));
+            } else if ($searchProblem == 'error') {
+                echo $html->tag('h2', format(__('An error occurred while fetching random sentences. 
+                        If this persists, please <a href="{}">let us know</a>.', true),
```
Same comment here.

jiru commented on app/views/sentences/several_random_sentences.ctp:
```diff
@@ -26,7 +26,7 @@
+if(!isset($searchProblem)) {
 ?>
 <div id="annexe_content">
     <div class="module">
```
The annexe content should be left visible regardless of the error, so that users can click the button to try again.

kwikadi commented:
Done :)

jiru commented:
Another minor problem: the h2 tag is not appropriate everywhere you used it. A simple p should be enough. If you want to make it bigger, style it with a CSS class.

jiru commented on app/views/sentences/several_random_sentences.ctp:
```diff
@@ -90,21 +89,43 @@ $this->set('title_for_layout', $pages->formatTitle(__('Random sentences', true))
+    <div id="main_content">
+        <div class="module">
+        <?php  
+            $this->Html->tag('h2', 'Random sentences', null);
```
Up to here, the code can be taken out of the `if()` block to avoid duplication with the corresponding `else`.

jiru commented on app/controllers/sentences_controller.php:
```diff
@@ -1008,10 +1022,19 @@ public function several_random_sentences()
+                    $this->set('searchProblem', 'disabled');
+                } else {
+                    $this->set('searchProblem', 'error');
+                }
```
Strange indentation here

jiru commented on app/views/sentences/several_random_sentences.ctp:
```diff
@@ -89,22 +88,30 @@ $this->set('title_for_layout', $pages->formatTitle(__('Random sentences', true))
+        } else if ($searchProblem == 'error') {
+            echo $html->tag('p', format(__('An error occurred while fetching random sentences. '.
+                    'If this persists, please <a href="{}">let us know</a>.', true),
+                $html->url(array("controller"=>"pages", "action" => "contact"))
```
Please fix indentation here too.

jiru commented on app/views/sentences/show.ctp:
```diff
@@ -233,4 +234,23 @@ $navigation->displaySentenceNavigation(
+            } else if ($searchProblem == 'error') {
+                echo $html->tag('p', format(__('An error occurred while fetching random sentences. '. 
+                        'If this persists, please <a href="{}">let us know</a>.', true),
+                    $html->url(array("controller"=>"pages", "action" => "contact"))
```
And here too.
